tools/schemapi/vega_expr.py

```diff
@@ -0,0 +1,674 @@
+                others = self.parameters[idx + 1 :]
+                if not others:
+                    return False
+                else:
+                    return any(sp.required for sp in others)
+
+        return False
+
+    def is_keyword(self) -> bool:
+        return keyword.iskeyword(self.name)
+
+    def is_incompatible_override(self) -> bool:
+        """
+        ``self.name_safe`` shadows an unrelated ``SchemaBase`` method.
+
+        Requires an ignore comment for a type checker.
+        """
+        return self.name_safe in _SCHEMA_BASE_MEMBERS
+
+    def is_variadic(self) -> bool:
+        """Position-only parameter separator `"/"` not allowed after `"*"` parameter."""
+        return self.is_overloaded() or any(p.variadic for p in self.parameters)
+
+    def __iter__(self) -> Iterator[Token]:
+        yield from self._children
+
+    @overload
+    def __getitem__(self, index: int) -> Token: ...
+    @overload
+    def __getitem__(self, index: slice) -> Sequence[Token]: ...
+    def __getitem__(self, index: int | slice) -> Token | Sequence[Token]:
+        return self._children.__getitem__(index)
+
+
+@dataclasses.dataclass
+class VegaExprParam:
+    name: str
+    required: bool
+    variadic: bool = False
+
+    @staticmethod
+    def star_args() -> LiteralString:
+        return f"{STAR_ARGS}: Any"
+
+    def to_str(self) -> str:
+        """Return as an annotated parameter, with a default if needed."""
+        if self.required:
+            return f"{self.name}: {INPUT_ANNOTATION}"
+        elif not self.variadic:
+            return f"{self.name}: {INPUT_ANNOTATION} = {NONE}"
+        else:
+            return self.star_args()
+
+    @classmethod
+    def iter_params(cls, raw_texts: Iterable[str], /) -> Iterator[Self]:
+        """Yields an ordered parameter list."""
+        is_required: bool = True
+        for s in raw_texts:
+            if s not in {"(", ")"}:
+                if s == "[":
+                    is_required = False
+                    continue
+                elif s == "]":
+                    is_required = True
+                    continue
+                elif s.isalnum():
+                    yield cls(s, required=is_required)
+                elif s == "...":
+                    yield cls(STAR_ARGS, required=False, variadic=True)
+                else:
+                    continue
+
+
+def parse_expressions(url: str, /) -> Iterator[tuple[str, VegaExprNode]]:
+    """Download, read markdown and iteratively parse into signature representations."""
+    for tok in read_tokens(download_expressions_md(url)):
+        if (
+            (children := tok.get(CHILDREN)) is not None
+            and (child := next(iter(children)).get(RAW)) is not None
+            and (match := FUNCTION_DEF_LINE.match(child))
+        ):
+            node = VegaExprNode(match[1], children)
+            if node.is_callable():
+                yield node.name, node.with_parameters().with_doc()
```

Comment:
# TODO
- Perform a first-pass to collect each `VegaExprNode.with_parameters()`
- Store each `VegaExprNode.name_safe` in a `ClassVar`
- Rewrite `VegaExprNode._doc_post_process`
	- This part should replace using **all** of the remapped names, instead of just the current one
	- See [timeUnitSpecifier](https://vega.github.io/vega/docs/expressions/#timeUnitSpecifier) which needs to know about `alt.expr.timeFormat`


	```py
	with_alt_references = re.sub(rf"({self.name}\()", f"alt.expr.{self.name_safe}(", highlight_params)
	```
- Then call `VegaExprNode.with_doc()` on each

Reply:
Fixed in [feat: Replace all Vega function refs in code blocks](https://github.com/vega/altair/pull/3600/commits/f6cc7762a9d68ad592868eaf963328eada07631c)